Engine/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -102,8 +102,15 @@ void WindowsWindow::Init(const WindowProps& props)
         glfwSetErrorCallback(GLFWErrorCallback);
     }
 
+// TODO: avoid using macros here
+#ifdef __APPLE__
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+#else
     glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
     glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 5);
+#endif
+
```

Comment:
This RHI implementation details leaks into window creation code - the opengl version that we better provide to glfw hints.

What do you think we can do to fix this, assuming other backends like Vulkan (or even DirectX) can be added to the mix with other hints values?

Reply:
Moved this to new function "GraphicsContext::PreInit" as glfwWindowHint must be called before the window is created